tools/validators/instance_validator/validate/generate_universe.py

```diff
@@ -23,23 +23,33 @@
 from yamlformat.validator import presubmit_validate_types_lib
 from yamlformat.validator import namespace_validator
 
-def BuildUniverse():
+def BuildUniverse(modified_types_filepath=None):
   """Generates the ontology universe.
 
+  Args:
+    modified_types_filepath: filepath to the modified ontology types
+
   Returns:
     Generated universe object.
   """
 
-  ontology_validator_exists = path.exists(path.join(
-      '..', 'ontology_validator'))
-  ontology_exists = path.exists(path.join('..', '..', '..', 'ontology'))
+  if modified_types_filepath:
+    modified_types_filepath = path.expanduser(modified_types_filepath)
+    modified_ontology_exists = path.exists(modified_types_filepath)
 
-  if not (ontology_validator_exists and ontology_exists):
-    print('ERROR: ontology validator or ontology have changed locations')
-    return None
+    if not modified_ontology_exists:
+      print('Specified filepath for modified ontology does not exist')
+      return None
+
+    external_file_lib.Validate(filter_text=None,
+                               original_directory=modified_types_filepath,
```

Comment:
actually, you can leave the original ontology here ` original_directory= ` and put the modified here `changed_directory=`
This way the validator will tell you if the change is still compatible with the origianl

Reply:
Done